Tune up codespell configuration to skip trough and not modify versioneer file

yarikoptic opened the pull request:
you don't have CI for codespell. If you want an easy github workflow -- let me know and I will add it so no typo sneaks in

smoia commented:
@yarikoptic could you add codespell to pre-commit please? In fact, if you could do it for all our packages, that'd be awesome!

yarikoptic commented:
- rebased
- added global exclude for pre-commit of vendored `physioqc/_version.py` (you don't want to fix it up, do you?)
- to add .pre-commit codespell stub I improved my https://github.com/yarikoptic/improveit/blob/HEAD/codespellit so now others should be able to use `codespellit add-precommit` ;-)
